api/opds_for_distributors.py

```diff
@@ -300,57 +284,65 @@ def fulfill(
         """Retrieve a bearer token that can be used to download the book.
 
-        :param kwargs: A container for arguments to fulfill()
-           which are not relevant to this vendor.
-
         :return: a FulfillmentInfo object.
         """
+        if (
+            delivery_mechanism.delivery_mechanism.drm_scheme
+            != DeliveryMechanism.BEARER_TOKEN
+        ):
+            raise DeliveryMechanismError(
+                "Cannot fulfill a loan through OPDS For Distributors using a delivery mechanism with DRM scheme %s"
+                % delivery_mechanism.delivery_mechanism.drm_scheme
+            )
 
         links = licensepool.identifier.links
+
         # Find the acquisition link with the right media type.
+        url = None
         for link in links:
             media_type = link.resource.representation.media_type
             if (
                 link.rel == Hyperlink.GENERIC_OPDS_ACQUISITION
-                and media_type == internal_format
+                and media_type == delivery_mechanism.delivery_mechanism.content_type
             ):
                 url = link.resource.representation.url
+                break
 
-                # Obtain a Credential with the information from our
-                # bearer token.
-                _db = Session.object_session(licensepool)
-                credential = self._get_token(_db)
-
-                # Build a application/vnd.librarysimplified.bearer-token
-                # document using information from the credential.
-                now = utc_now()
-                expiration = int((credential.expires - now).total_seconds())  # type: ignore[operator]
-                token_document = dict(
-                    token_type="Bearer",
-                    access_token=credential.credential,
-                    expires_in=expiration,
-                    location=url,
-                )
+        if url is None:
+            # We couldn't find an acquisition link for this book.
+            raise CannotFulfill()
 
-                return FulfillmentInfo(
-                    licensepool.collection,
-                    licensepool.data_source.name,
-                    licensepool.identifier.type,
-                    licensepool.identifier.identifier,
-                    content_link=None,
-                    content_type=DeliveryMechanism.BEARER_TOKEN,
-                    content=json.dumps(token_document),
-                    content_expires=credential.expires,
-                )
+        # Obtain a Credential with the information from our
+        # bearer token.
+        _db = Session.object_session(licensepool)
+        credential = self._get_token(_db)
 
-        # We couldn't find an acquisition link for this book.
-        raise CannotFulfill()
+        # Build a application/vnd.librarysimplified.bearer-token
+        # document using information from the credential.
+        now = utc_now()
+        assert credential.expires is not None
```

Comment:
Does an AssertionError in production lead to an INTERNAL_SERVER_ERROR problem detail?